app/services/finance/stripe_payment_intent_update_service.rb

```diff
@@ -1,13 +1,13 @@
 # frozen_string_literal: true
 
 class Finance::StripePaymentIntentUpdateService
-  def initialize(provider_account, stripe_event)
-    @stripe_event = stripe_event
-    @payment_intent_data = stripe_event.data.object
-    @payment_intent = PaymentIntent.by_invoice(provider_account.buyer_invoices).find_by!(reference: payment_intent_data['id'])
+  def initialize(provider_account, source_object)
+    @source_object = source_object.to_hash
+    @payment_intent_data = @source_object.dig(:data, :object) || @source_object
```

Comment:
Why is in some cases `@payment_intent_data == @source_object`? Not really obvious from here

Reply:
That is because of https://github.com/3scale/porta/pull/2350/commits/4110d34e3da592fcfe8a0cae83744e2be3caf0cc. The Stripe webhook sends a `Stripe::Event`, while our own callback after the checkout page sends a `Hash` of just the part of the payment intent (without being wrapped within a Stripe event).